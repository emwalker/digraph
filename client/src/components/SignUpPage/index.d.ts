import React, { Component } from 'react';
type State = {
    gCaptchaResponse: string | null;
};
type Props = {};
declare class SignUpPage extends Component<Props, State> {
    constructor(props: Props);
    onCaptchaSumit: (token: string | null) => void;
    renderCreateAccountButton: () => React.JSX.Element;
    render: () => React.JSX.Element;
}
export default SignUpPage;
