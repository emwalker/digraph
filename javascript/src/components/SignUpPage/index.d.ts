import { Component } from 'react';
declare type State = {
    gCaptchaResponse: string | null;
};
declare type Props = {};
declare class SignUpPage extends Component<Props, State> {
    constructor(props: Props);
    onCaptchaSumit: (token: string | null) => void;
    renderCreateAccountButton: () => JSX.Element;
    render: () => JSX.Element;
}
export default SignUpPage;
