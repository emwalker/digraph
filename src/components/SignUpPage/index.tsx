import React, { Component } from 'react'
import ReCAPTCHA from 'react-google-recaptcha'

import GithubLogin from 'components/ui/GithubLogin'

/* eslint react/no-string-refs: 0 */

type State = {
  gCaptchaResponse: string | null,
}

type Props = {}

// http://emumba.com/blog/2016-12-07-setting-up-google-recaptcha-in-a-reactjs-app/

class SignUpPage extends Component<Props, State> {
  constructor(props: Props) {
    super(props)
    this.state = {
      gCaptchaResponse: null,
    }
  }

  onCaptchaSumit = (token: string | null) => {
    this.setState({ gCaptchaResponse: token })
  }

  renderCreateAccountButton = () => (
    <button
      disabled={!this.state.gCaptchaResponse}
      className="btn btn-primary"
      type="submit"
    >
      Create an account
    </button>
  )

  render = () => (
    <div className="SignUpPage clearfix">
      <div className="col-6">
        <h1>Join Digraph</h1>
        <p className="lead">
          {'Keep track of everything you\'ve ever read on the Internet.'}
        </p>

        <GithubLogin className="mb-5" />

        <h2 className="f2-light mb-1">Or create an account</h2>

        <form>
          <dl className="form-group required">
            <dt><label htmlFor="username">Username</label></dt>
            <dd>
              <input id="username" className="form-control" type="text" />
              <p className="note">Used similarly to a GitHub username.</p>
            </dd>
          </dl>
          <dl className="form-group required">
            <dt><label htmlFor="email">Email address</label></dt>
            <dd>
              <input id="email" className="form-control" type="text" />
              <p className="note">
                For getting in touch with you if something important comes up. We will not
                sell your email address to third parties.
              </p>
            </dd>
          </dl>
          <dl className="form-group required">
            <dt><label htmlFor="password">Password</label></dt>
            <dd>
              <input id="password" className="form-control" type="text" />
              <p className="note">
                At least 15 characters long.
              </p>
            </dd>
          </dl>
        </form>

        <h2 className="mb-2">Verify account</h2>
        <div className="width-full mb-4">
          <ReCAPTCHA
            ref="recaptcha"
            sitekey="6LfOQosUAAAAAClaTyOmN0d64syvFjxdys1nUkbd"
            onChange={this.onCaptchaSumit}
          />
        </div>

        {this.renderCreateAccountButton()}
      </div>
    </div>
  )
}

export default SignUpPage
