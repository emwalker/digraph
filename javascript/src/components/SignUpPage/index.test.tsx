import React from 'react'
import { shallow } from 'enzyme'

import SignUpPage from './index'

const props = {}

describe('<SignUpPage />', () => {
  const wrapper = shallow(<SignUpPage {...props} />)

  it('renders', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
