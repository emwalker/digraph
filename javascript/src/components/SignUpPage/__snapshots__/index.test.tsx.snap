// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpPage /> renders 1`] = `
<div
  className="SignUpPage clearfix"
>
  <div
    className="col-6"
  >
    <h1>
      Join Digraph
    </h1>
    <p
      className="lead"
    >
      Keep track of everything you've ever read on the Internet.
    </p>
    <GithubLogin
      className="mb-5"
    />
    <h2
      className="f2-light mb-1"
    >
      Or create an account
    </h2>
    <form>
      <dl
        className="form-group required"
      >
        <dt>
          <label
            htmlFor="username"
          >
            Username
          </label>
        </dt>
        <dd>
          <input
            className="form-control"
            id="username"
            type="text"
          />
          <p
            className="note"
          >
            Used similarly to a GitHub username.
          </p>
        </dd>
      </dl>
      <dl
        className="form-group required"
      >
        <dt>
          <label
            htmlFor="email"
          >
            Email address
          </label>
        </dt>
        <dd>
          <input
            className="form-control"
            id="email"
            type="text"
          />
          <p
            className="note"
          >
            For getting in touch with you if something important comes up. We will not sell your email address to third parties.
          </p>
        </dd>
      </dl>
      <dl
        className="form-group required"
      >
        <dt>
          <label
            htmlFor="password"
          >
            Password
          </label>
        </dt>
        <dd>
          <input
            className="form-control"
            id="password"
            type="text"
          />
          <p
            className="note"
          >
            At least 15 characters long.
          </p>
        </dd>
      </dl>
    </form>
    <h2
      className="mb-2"
    >
      Verify account
    </h2>
    <div
      className="width-full mb-4"
    >
      <AsyncScriptLoader(ReCAPTCHA)
        onChange={[Function]}
        sitekey="6LfOQosUAAAAAClaTyOmN0d64syvFjxdys1nUkbd"
      />
    </div>
    <button
      className="btn btn-primary"
      disabled={true}
      type="submit"
    >
      Create an account
    </button>
  </div>
</div>
`;
